src/Symfony/Component/Uid/Uuid.php

```diff
@@ -124,15 +124,17 @@ final public static function v8(string $uuid): UuidV8
     /**
      * @param int-mask-of<Uuid::FORMAT_*> $format
      */
-    public static function isValid(string $uuid /*, int $format = self::FORMAT_RFC_4122 */): bool
+    public static function isValid(string $uuid /* , int $format = self::FORMAT_RFC_4122 */): bool
     {
         $format = 1 < \func_num_args() ? func_get_arg(1) : self::FORMAT_RFC_4122;
 
         if (36 === \strlen($uuid) && !($format & self::FORMAT_RFC_4122)) {
             return false;
         }
 
-        $uuid = self::transformToRfc4122($uuid, $format);
+        if (!$uuid = self::transformToRfc4122($uuid, $format)) {
```

Comment:
here I'd do a strict check because the type is the decider, not the value
```suggestion
        if (false === $uuid = self::transformToRfc4122($uuid, $format)) {
```

Reply:
Agree